test/unit_CartesianProduct.jl

```diff
@@ -273,3 +273,19 @@ for N in [Float64, Float32, Rational{Int}]
           EmptySet
     @test b × e == e × b == cpa × e == e × cpa == e × e == e
 end
+
+for N in [Float64, Float32]
+    #is_intersection_empty
+    i1 = Interval(N[0, 1])
+    i2 = Interval(N[2, 3])
+    h1 = Hyperrectangle(low=N[3, 4], high=N[5, 7])
+    h2 = Hyperrectangle(low=N[5, 5], high=N[6, 8])
+    cpa1 = CartesianProductArray([i1, i2, h1])
+    cpa2 = CartesianProductArray([i1, i2, h2])
+    G = HPolyhedron([HalfSpace(N[1, 0, 0, 0], N(1))])
+    G_empty = HPolyhedron([HalfSpace(N[1, 0, 0, 0], N(-1))])
```

Comment:
If you do the change to `AbstractPolyhedron`, this wrapping in `HPolyhedron` is obsolete.

Reply:
I am getting an error:
```
ERROR: MethodError: LazySets.is_intersection_empty(::CartesianProductArray{Float64,AbstractHyperrectangle{Float64}}, ::HPolyhedron{Float64}) is ambiguous. Candidates:
  is_intersection_empty(X::CartesianProductArray{N,S} where S<:LazySet{N}, Y::AbstractPolyhedron{N}) where N<:Real in LazySets at /Users/kpotomkin/.julia/dev/LazySets/src/is_intersection_empty.jl:1387
  is_intersection_empty(X::CartesianProductArray{N,S}, Y::AbstractPolyhedron{N}) where {N<:Real, S<:LazySet{N<:Real}} in LazySets at /Users/kpotomkin/.julia/dev/LazySets/src/is_intersection_empty.jl:1387
  is_intersection_empty(X::LazySet{N}, P::Union{AbstractPolytope{N}, HPolyhedron{N}}) where N<:Real in LazySets at /Users/kpotomkin/.julia/dev/LazySets/src/is_intersection_empty.jl:973
```

Reply:
The first two lines look like you define several methods that do the same. Still, you may have to add more methods to resolve the ambiguities. If you want I can do the generalization to `AbstractPolyhedron` in a follow-up PR.